workers.tf

```diff
@@ -24,33 +51,117 @@ resource "aws_autoscaling_group" "workers" {
+  security_groups             = ["${local.worker_security_group_id}", "${var.worker_additional_security_group_ids}", "${compact(split(",",lookup(var.launch_configuration_worker_groups[count.index],"additional_security_group_ids", local.workers_group_defaults["additional_security_group_ids"])))}"]
+  iam_instance_profile        = "${element(aws_iam_instance_profile.launch_configuration_workers.*.id, count.index)}"
+  image_id                    = "${lookup(var.launch_configuration_worker_groups[count.index], "ami_id", local.workers_group_defaults["ami_id"])}"
+  instance_type               = "${lookup(var.launch_configuration_worker_groups[count.index], "instance_type", local.workers_group_defaults["instance_type"])}"
+  key_name                    = "${lookup(var.launch_configuration_worker_groups[count.index], "key_name", local.workers_group_defaults["key_name"])}"
+  user_data_base64            = "${base64encode(element(data.template_file.launch_configuration_userdata.*.rendered, count.index))}"
+  ebs_optimized               = "${lookup(var.launch_configuration_worker_groups[count.index], "ebs_optimized", lookup(local.ebs_optimized, lookup(var.launch_configuration_worker_groups[count.index], "instance_type", local.workers_group_defaults["instance_type"]), false))}"
+  enable_monitoring           = "${lookup(var.launch_configuration_worker_groups[count.index], "enable_monitoring", local.workers_group_defaults["enable_monitoring"])}"
+  spot_price                  = "${lookup(var.launch_configuration_worker_groups[count.index], "spot_price", local.workers_group_defaults["spot_price"])}"
+  placement_tenancy           = "${lookup(var.launch_configuration_worker_groups[count.index], "placement_tenancy", local.workers_group_defaults["placement_tenancy"])}"
+  count                       = "${var.launch_configuration_worker_group_count}"
 
   lifecycle {
     create_before_destroy = true
   }
 
   root_block_device {
-    volume_size           = "${lookup(var.worker_groups[count.index], "root_volume_size", local.workers_group_defaults["root_volume_size"])}"
-    volume_type           = "${lookup(var.worker_groups[count.index], "root_volume_type", local.workers_group_defaults["root_volume_type"])}"
-    iops                  = "${lookup(var.worker_groups[count.index], "root_iops", local.workers_group_defaults["root_iops"])}"
+    volume_size           = "${lookup(var.launch_configuration_worker_groups[count.index], "root_volume_size", local.workers_group_defaults["root_volume_size"])}"
+    volume_type           = "${lookup(var.launch_configuration_worker_groups[count.index], "root_volume_type", local.workers_group_defaults["root_volume_type"])}"
+    iops                  = "${lookup(var.launch_configuration_worker_groups[count.index], "root_iops", local.workers_group_defaults["root_iops"])}"
     delete_on_termination = true
   }
 }
 
+resource "aws_iam_instance_profile" "launch_configuration_workers" {
```

Comment:
Do we need a second instance profile? Can't both LC and LT type ASGs use the same one?

Reply:
Ahh, yes, I think it needs to stay.

Reply:
Yep, especially since this change will support launching with both types.